internal/storage/v2/grpc/trace_storage.proto

```diff
@@ -83,4 +124,27 @@ service TraceReader {
   // GetOperations returns all operation names for a given service
   // known to the backend from traces within its retention period.
   rpc GetOperations(GetOperationsRequest) returns (GetOperationsResponse) {}
+
+  // FindTraces returns a stream that retrieves traces matching query parameters.
+  //
+  // The chunking rules are the same as for GetTraces.
+  //
+  // If no matching traces are found, an empty stream is returned.
+  // If an error is encountered, the iterator returns the error and stops.
+  //
+  // There's currently an implementation-dependent ambiguity whether all query filters
+  // (such as multiple tags) must apply to the same span within a trace, or can be satisfied
+  // by different spans.
+  rpc FindTraces(TraceQueryParams) returns (stream opentelemetry.proto.trace.v1.TracesData) {}
```

Comment:
@yurishkuro I'm passing in `TraceQueryParams` directly here - do we want a layer of indirection?

Reply:
Yes. This API should be nearly identical to api_v3.